mabwiser/approximate.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: Apache-2.0
+
+from collections import defaultdict
+from copy import deepcopy
+from itertools import chain
+from typing import List, NoReturn, Optional, Union
+
+import numpy as np
+from joblib import Parallel, delayed
+
+from mabwiser.greedy import _EpsilonGreedy
+from mabwiser.linear import _Linear
+from mabwiser.neighbors import _Neighbors
+from mabwiser.popularity import _Popularity
+from mabwiser.rand import _Random
+from mabwiser.softmax import _Softmax
+from mabwiser.thompson import _ThompsonSampling
+from mabwiser.ucb import _UCB1
+from mabwiser.utils import Arm, _BaseRNG, create_rng
+
+
+class _ApproximateNeighbors(_Neighbors):
+
+    def fit(self, decisions: np.ndarray, rewards: np.ndarray, contexts: np.ndarray = None) -> NoReturn:
+        # Initialize planes
+        self._initialize(contexts.shape[1])
+
+        # Set the historical data for prediction
+        self.decisions = decisions
+        self.contexts = contexts
+
+        # Binarize the rewards if using Thompson Sampling
+        if isinstance(self.lp, _ThompsonSampling) and self.lp.binarizer:
+            self.rewards = self._binarize_ts_rewards(decisions, rewards)
+        else:
+            self.rewards = rewards
+
+        # Fit hashes for each training context
+        self._fit_operation(contexts)
+
+    def partial_fit(self, decisions: np.ndarray, rewards: np.ndarray,
+                    contexts: Optional[np.ndarray] = None) -> NoReturn:
+
+        # Binarize the rewards if using Thompson Sampling
+        if isinstance(self.lp, _ThompsonSampling) and self.lp.binarizer:
+            rewards = self._binarize_ts_rewards(decisions, rewards)
+
+        # Add more historical data for prediction
+        self.decisions = np.concatenate((self.decisions, decisions))
+        self.rewards = np.concatenate((self.rewards, rewards))
+        self.contexts = np.concatenate((self.contexts, contexts))
+
+        # Fit hashes for each training context
+        self._fit_operation(contexts)
+
+    def _get_neighbors(self, row_2d):
```

Comment:
Why not make these abstract methods using the ABC class? Would be more explicit.